### PR TITLE
Test against Latest WP Version instead of Trunk

### PR DESCRIPTION
**Pull request by jarednova**
**Ticket**: #2562 

## Issue
There appears to be an unrelated failure occurring due to `trunk` changes in WP on #2562. This begs the question — why test against trunk at all?

## Solution
Swap `trunk` for `latest`

## Impact
We'll be a bit behind should a breaking change come in WP's functions (whereas before we could have caught it before formal release). However, 

## Usage Changes
None

## Considerations
Should we maybe keep one trunk test in the matrix and allow failures?

### CODE REVIEW

**Comment by nlemoine** [comment]
Testing `trunk`, as well as latest PHP version is important to spot bugs before they are released (see https://github.com/timber/timber/issues/2548) which prevents from running into issues when it comes out.

I don't have stats on this but I feel that latest version adoption is quite fast nowadays.

**Comment by jarednova** [comment]
I hear that @nlemoine — maybe I was being a bit too cavalier. I agree, we should preserve some amount of `trunk` tests. 

Perhaps the better "middle way" is now here with the latest commit: test against `latest` for most of the tests and keep one set of `trunk` tests. This way we can better identify what is a live issue in the code against WP vs. what could be a coming issue (or an accidental regression that the WP core team will likely resolve). What do you think?

**Comment by nlemoine** [comment]
I would keep two `trunk`: 
- PHP 8.1/trunk: to ensure Timber is still working on edge configuration
- PHP 7.4/trunk: to ensure Timber is still working on edge WP

Both with allow failure on, so we can keep an eye on bugs to come but still let us merge things that are not ready yet to pass (or waiting that WP fixes some issues).

**Comment by nlemoine** [comment]
I've updated the test configuration. 

I also think we should support PHP 8.1 as it's already out since a few months. It would be a shame to have a PHP 8.0.x only constraint. I already fixed the issues I've found for that version and It's only a matter of time before we have to support this.

**Comment by gchtr** [comment]
So how do we manage the failed runs against WP trunk? Is there a state that they pass with the errors? Or do we manually check if it’s only those tests that fail and then overrule merging?



In the current test run, there’s also a PCOV error with [PHP 7.4 against WP latest](https://github.com/timber/timber/runs/5728064459?check_suite_focus=true).

**Comment by nlemoine** [comment]
I don't think this related to GA trunk tests failing? It says "Review required"

**Review by gchtr — APPROVED** [review]

**Comment by gchtr** [comment]
> I don't think this is related to GA trunk tests failing? It says "Review required"

😅  Now I feel stupid. Merging isn’t blocked now, but I think we still need to manually check whether any failing checks are against WP trunk before merging.

**Comment by nlemoine** [comment]
> I think we still need to manually check whether any failing checks are against WP trunk before merging.

Not sure I understand your comment. If some failures happen on trunk, you will see them but they will not prevent the "Merge" button. A failure on non trunk test will disable the "Merge" button. 

Capture d’écran 2022-04-09 à 09 31 09

**Comment by jarednova** [comment]
This works! So just to clarify/confirm, the _review_ is what creates the permission for the merge — not a setting in the config or matrix

**Pull request merged**
